### ISSUE TITLE
Why there is no square root at area_temperature?

### ISSUE THREAD

**Issue by jiminbot20** [tensorflow/tensor2tensor#1900]
https://github.com/tensorflow/tensor2tensor/blob/5623deb79cfcd28f8f8c5463b58b5bd76a81fd0d/tensor2tensor/layers/area_attention.py#L415

In typical dot product attention, logit which is the input matrix of softmax supposed to be divided by square rooted temperature like the equation below.
![image](https://user-images.githubusercontent.com/63258184/140316988-7c87e78f-0ad8-415d-a860-8d751c124cb0.png)

However, in this code, logit is just divided with temperature without a square root. Is it correct or wrong? If it is correct, could you explain why you didn't add square root?
